src/common/com/intellij/plugins/haxe/util/HaxeResolveUtil.java

```diff
@@ -449,11 +450,14 @@ public static HaxeClassResolveResult getHaxeClassResolveResult(@Nullable PsiElem
           return result;
         }
         // try iterator
-        result = getHaxeClassResolveResult(resolveResultHaxeClass == null ? null : resolveResultHaxeClass.findHaxeMethodByName("iterator"),
-                                           resolveResult.getSpecialization());
-        return result.getSpecialization().containsKey(null, "T")
-               ? result.getSpecialization().get(null, "T")
-               : HaxeClassResolveResult.EMPTY;
```

Comment:
This change is the one that allows us to figure out the type of a variable declared in a for loop.  Instead of just looking for an iterator and assuming the type variable name, we now resolve the iterator type properly, and then go check its next() method to get the proper type.